api/src/pcapi/core/offers/api.py

```diff
@@ -145,14 +146,14 @@ def create_offer(
     user: User,
     save_as_active: bool = True,
 ) -> Offer:
-    subcategory = subcategories.ALL_SUBCATEGORIES_DICT.get(offer_data.subcategory_id)  # type: ignore [arg-type]
     venue = load_or_raise_error(Venue, offer_data.venue_id)
     check_user_has_access_to_offerer(user, offerer_id=venue.managingOffererId)  # type: ignore [attr-defined]
     _check_offer_data_is_valid(offer_data, offer_data.is_educational)  # type: ignore [arg-type]
-    if _is_able_to_create_book_offer_from_isbn(subcategory):  # type: ignore [arg-type]
+    subcategory = _get_subcategory_or_raise(offer_data.subcategory_id)
```

Comment:
On vient d'appeler `_check_offer_data_is_valid()` qui a déjà vérifié que la sous-catégorie fournie existe bien. On peut donc simplement écrire : 

```suggestion
    subcategory = subcategories.ALL_SUBCATEGORIES_DICT[offer_data.subcategory_id]
```